### PR TITLE
Support omise api 2019

### PR DESCRIPTION
**Pull request by AnasNaouchi**
## Description

Update request models and unit tests to support the latest 2019 omise-api version.

### CODE REVIEW

File: Omise/Models/ChargeRequest.cs [hunk 1]
```diff
@@ -6,28 +6,35 @@ namespace Omise.Models
...
-        public string ReturnUri { get; set; }
+        public string? ReturnUri { get; set; }
         [JsonProperty("platform_fee")]
-        public PlatformFeeRequest PlatformFee { get; set; }
+        public PlatformFeeRequest? PlatformFee { get; set; }
+        [JsonProperty("first_charge")]
+        public string? FirstCharge { get; set; }
+         public string? Ip { get; set; }
+         [JsonProperty("linked_account")]
+         public string? LinkedAccount { get; set; }
```
Comment (aashishgurung) [inline Omise/Models/ChargeRequest.cs:31]: I think this needs to be an object.

> The linked account object for direct debit.
> - https://docs.opn.ooo/charges-api

File: Omise/Resources/StatusSpecificDisputeResource.cs [hunk 2]
```diff
@@ -19,10 +19,10 @@ static string pathForStatus(DisputeStatus status)
                     return "/disputes/open";
                 case DisputeStatus.Pending:
                     return "/disputes/pending";
-                case DisputeStatus.Closed:
                 case DisputeStatus.Won:
+                    return "/disputes/won";
                 case DisputeStatus.Lost:
-                    return "/disputes/closed";
+                    return "/disputes/lost";
```
Comment (aashishgurung) [inline Omise/Resources/StatusSpecificDisputeResource.cs]: I don't think we have `/disputes/lost` endpoint. It should be `/disputes/closed`

https://docs.opn.ooo/disputes-api#closed

**Review by timothyOmise — APPROVED** [review]

File: Omise/Client.cs [hunk 3]
```diff
@@ -1,87 +1,87 @@
...
+
+        public IEnvironment Environment
+        {
+            get { return environment; }
+        }
+
+        public string APIVersion
+        {
+            get { return requester.APIVersion; }
+            set { requester.APIVersion = value; }
```
Comment (aashishgurung) [inline Omise/Client.cs]: Do we still need `set` method for APIVersion?

Comment (AnasNaouchi) [reply Omise/Client.cs]: You are correct, we should remove `set`.

**Review by aashishgurung — APPROVED** [review]

**Pull request merged**